presto-native-execution/presto_cpp/main/http/HttpClient.cpp

```diff
@@ -305,31 +299,25 @@ class ConnectionHandler : public proxygen::HTTPConnector::Callback {
       std::chrono::milliseconds connectTimeout,
       folly::EventBase* eventBase,
       const folly::SocketAddress& address,
-      const std::string& clientCertAndKeyPath,
-      const std::string& ciphers)
+      folly::SSLContextPtr sslContext)
       : responseHandler_(responseHandler),
         sessionPool_(sessionPool),
         transactionTimer_(transactionTimeout),
         connectTimeout_(connectTimeout),
         eventBase_(eventBase),
         address_(address),
-        clientCertAndKeyPath_(clientCertAndKeyPath),
-        ciphers_(ciphers) {}
+        sslContext_(std::move(sslContext)) {}
 
-  bool useHttps() {
-    return !(clientCertAndKeyPath_.empty() && ciphers_.empty());
+  bool useHttps() const {
+    return sslContext_ != nullptr;
   }
 
   void connect() {
     connector_ =
         std::make_unique<proxygen::HTTPConnector>(this, transactionTimer_);
     if (useHttps()) {
-      auto context = std::make_shared<folly::SSLContext>();
-      context->loadCertKeyPairFromFiles(
-          clientCertAndKeyPath_.c_str(), clientCertAndKeyPath_.c_str());
-      context->setCiphersOrThrow(ciphers_);
       connector_->connectSSL(
-          eventBase_, address_, context, nullptr, connectTimeout_);
+          eventBase_, address_, sslContext_, nullptr, connectTimeout_);
```

Comment:
```
useHttps()
```

Reply:
Yeah `useHttps()` is called in the branch.